addons/product/static/src/js/product_pricelist_report.js

```diff
@@ -1,317 +1,243 @@
+
+
+export class ProductPricelistReport extends Component {
+    setup() {
+        this.action = useService("action");
+        this.orm = useService("orm");
+
         this.MAX_QTY = 5;
-    },
-
-    //--------------------------------------------------------------------------
-    // Handlers
-    //--------------------------------------------------------------------------
-
-    /**
-     * Add a quantity when add(+) button clicked.
-     *
-     * @private
-     */
-    _onClickAddQty: function () {
+        const pastState = this.props.state || {};
+
+        // Putting everything that needs to be reactive on the useState wrapper.
+
+        // While activeIds and activeModel are not necessarily reactive we need
+        // them to be keeped for the useSetupAction going back and forward on links.
+        this.state = useState({
+            activeIds: pastState.active_ids || this.props.action.context.active_ids,
+            activeModel: pastState.active_model || this.props.action.context.active_model,
+            displayPricelistTitle: pastState.displayPricelistTitle || false,
+            html: "",
+            pricelists: [],
+            quantities: pastState.quantities || [1, 5, 10],
+            selectedPricelist: {},
+        });
+
+        onWillStart(async () => {
+            this.state.pricelists = await this.getPricelists()
+            this.state.selectedPricelist = pastState.selectedPricelist || this.state.pricelists[0];
+
+            this.renderHtml();
+        });
+
+        onRendered(() => {
+            this.env.config.setDisplayName(this.displayName);
+        });
+
+        /*
+        When following the link of a product and coming back we need to keep the 
+        precedent state:
+            - if the pricelist was being showed
+            - wich pricelist is selected at the moment
+            - which quantities
+            - the products (or product variants) for which the report was opened
+            - if the active model is product.product or product.template
+        */
+        useSetupAction({
+            getLocalState: () => {
+                return {
+                    active_ids: this.active_ids,
+                    active_model: this.active_model,
+                    displayPricelistTitle: this.displayPricelistTitle,
+                    quantities: this.quantities,
+                    selectedPricelist: this.selectedPricelist,
+              };
+            },
+        });
+    }
+
+    get activeIds() {
+        return this.state.activeIds;
+    }
+
+    get activeModel() {
+        return this.state.activeModel;
+    }
+
+    get displayName() {
+        return this.env._t("Pricelist Report");
+    }
+
+    get displayPricelistTitle() {
+        return this.state.displayPricelistTitle;
+    }
+
+    get reportParams() {
+        return {
+            active_model: this.activeModel || 'product.template',
+            active_ids: this.state.activeIds || [],
```

Comment:
Why do the `active_model` fallbacks to `product.template`? What if it's a product, then the ids will be messed up... 
How can I try to test without having an `active_ids` or `active_model`? Is it possible?
```suggestion
            active_model: this.props.action.context.active_model || 'product.template',
            active_ids: this.props.action.context.active_ids || [],
```

Reply:
When you reload the page, the current state is lost. In the precedent code (before this change) the report fell back to default:
- model : product.template
-  ids : empty (load all templates)

I'm just imitating the behavior from before.